File: ts/project/.sdk/src/cmp/java/fragment/EntityListOp.fragment.java
package JAVAPACKAGE.entity;

import java.util.LinkedHashMap;
import java.util.Map;

import JAVAPACKAGE.core.Context;
import JAVAPACKAGE.core.SdkClient;

// Fragment carrier: only the section between the EJECT markers is spliced
// into the generated entity class (which extends EntityBase).
abstract class EntityListOpFragment extends EntityBase {

  EntityListOpFragment(SdkClient client, Map<String, Object> entopts) {
    super("entityname", client, entopts);
  }

// EJECT-START

  @Override
  public Object list(Map<String, Object> reqmatch, Map<String, Object> ctrl) {
    Map<String, Object> ctxmap = new LinkedHashMap<>();
    ctxmap.put("opname", "list");
    ctxmap.put("ctrl", ctrl);
    ctxmap.put("match", this.match);
    ctxmap.put("data", this.data);
    ctxmap.put("reqmatch", reqmatch);
    Context ctx = this.utility.makeContext.apply(ctxmap, this.entctx);

    return runOp(ctx, () -> {
      if (ctx.result != null) {
        if (ctx.result.resmatch != null) {
          this.match = ctx.result.resmatch;
        }
      }
    });
  }

// EJECT-END
}

File: ts/project/.sdk/src/cmp/scala/fragment/EntityListOp.fragment.scala
package SCALAPACKAGE.entity

import java.util.{LinkedHashMap, Map => JMap}
import SCALAPACKAGE.core.SdkClient

abstract class EntityListOpFragment(client: SdkClient, entopts: JMap[String, Object])
    extends EntityBase("entityname", client, entopts) {

// EJECT-START

  override def list(reqmatch: JMap[String, Object], ctrl: JMap[String, Object]): Object = {
    val ctxmap = new LinkedHashMap[String, Object]()
    ctxmap.put("opname", "list")
    ctxmap.put("ctrl", ctrl)
    ctxmap.put("match", this.matchState)
    ctxmap.put("data", this.dataState)
    ctxmap.put("reqmatch", reqmatch)
    val ctx = this.utility.makeContext(ctxmap, this.entctx)

    runOp(ctx, () => {
      if (ctx.result != null) {
        if (ctx.result.resmatch != null) this.matchState = ctx.result.resmatch
      }
    })
  }

// EJECT-END
}
